server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/builder/AbstractStarTreeBuilderTests.java

```diff
@@ -2356,6 +3085,24 @@ public void testMergeFlowWithMaxLeafDocs() throws IOException {
          */
         assertEquals(635, starTreeDocuments.size());
         validateStarTree(builder.getRootNode(), 4, sf.getStarTreeConfig().maxLeafDocs(), builder.getStarTreeDocuments());
+        metaOut.close();
+        dataOut.close();
+
+        StarTreeMetadata starTreeMetadata = new StarTreeMetadata(
```

Comment:
nit : Lets refactor all such methods to common method

Reply:
same comment as below resolving